src/core/reactor.cc

```diff
@@ -2218,6 +2219,45 @@ void reactor::kill(pid_t pid, int sig) {
     ret.throw_if_error();
 }
 
+future<group_details> reactor::grp_detail(std::string_view groupname) noexcept {
+    return futurize_invoke([groupname, this] {
+        return _thread_pool->submit<syscall_result_extra<struct group>>([groupname = sstring(groupname)] {
+            struct group *gg = ::getgrnam(groupname.c_str());
+            if (!gg) {
+                return wrap_syscall(-1, *gg); // Simulating failure
+            }
+            return wrap_syscall(0, *gg);
+        }).then([groupname = sstring(groupname)] (syscall_result_extra<struct group> gr) {
+            gr.throw_fs_exception_if_error("group failed", groupname);
```

Comment:
why you are throwing an `filesystem_error` if `getgrnam()` fails or the matching entry is not found, and `groupname`  does not evaluate to a pathname? probably a `system_error` is better here?

and, this design implies an assumption: if an entry is not found, an exception is thrown with an unknown `errno`. IMO, this is not ideal:

- no matched entry should not be considered as an exception. it's one of the outcomes the caller should handle, and the caller should not handle it in the error handling path.
- throwing an exception with an unknown `errno` is not acceptable.

both of these arguments could be controversial. but i think the bottom line is to get back to the caller with the correct message.

Reply:
Agree, how about create a structure something like:
```
struct group_details_value {
    char *group_name;
    char *group_password;
    uint64_t group_gid;
    char **group_member;
};

struct group_details {
    int error;
    struct group_details_value details;
};
```
And populate errno values to group_details.error and can let the caller handle the values when non zero, if not the caller can look for fields in group_details.details ? Thanks

Reply:
i'd suggest 

1. return an `optional<group_details>` 
2. reset `errno` before calling this API. this suggestion is proposed in another comment.

BTW, might want to rename `group_details` to `group` or `group_info`.